ariadne/cardano/test/backend/Test/Spec/AcidState.hs

```diff
@@ -0,0 +1,97 @@
+module Test.Spec.AcidState (spec) where
+
+import Control.Exception.Base (ErrorCall(..), handle)
+import Control.Monad.Component hiding (throwM)
+
+import System.IO.Temp
+
+import Data.Acid (closeAcidState, openLocalStateFrom, query)
+import Data.Acid.Local (getState)
+
+import qualified Data.ByteString.Lazy as BL
+import Data.Text.Buildable (build)
+import Data.Text.Lazy.Builder (toLazyText)
+
+import Test.Hspec (Spec, describe)
+import Test.Hspec.QuickCheck (prop)
+import Test.QuickCheck (arbitrary, withMaxSuccess)
+import Test.QuickCheck.Monadic (monadicIO, pick, run)
+
+import Test.Spec.CreateWallet (NewWallet, applyNewWallet, genNewWalletRq)
+import Test.Spec.Fixture (genSpendingPasswords, withLayerLocalStorage)
+
+import Ariadne.Wallet.Cardano.Kernel.DB.AcidState (DB(..), Snapshot(..), defDB)
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet (HdWallets(..))
+import Ariadne.Wallet.Cardano.Kernel.DB.Util.IxSet (IxSet, size)
+import Ariadne.Wallet.Cardano.Kernel.Internal (PassiveWallet(..), wallets)
+import Ariadne.Wallet.Cardano.WalletLayer (PassiveWalletLayer, pwlCreateWallet)
+
+spec :: Spec
+spec = describe "AcidState" $ do
+  prop "can open empty database" $ withMaxSuccess 1 $
+    monadicIO $ run $ withSystemTempDirectory "testWalletDBEmpty" $ \path -> do
+        getState path defDB 0 True >>= \case
+          Left _ -> return False
+          Right (acidDB, eventTag) -> do
+            db <- liftIO $ query acidDB Snapshot
+            let isEmptyDB = checkEmptyWalletDB db
+            return $ isEmptyDB && (eventTag == noEventsMsg)

```

Comment:
Well, this test is too low-level, essentially it tests internals of `acid-state`. I think it's better to use `withLayerLocalStorage` like you do in the second test. `withLayerLocalStorage` calls `openLocalStateFrom` which is actually used in production, while `getState` is some internal function which should be used directly only in very rare cases.

Reply:
Ok. Done.